### PR TITLE
Add deployment environment API endpoint

### PR DESCRIPTION
**Pull request by dotNomad**
This PR adds a new endpoint for our binary `GET /api/deployments/$NAME/environment` which will either:
- Fail in a myriad of different ways
- Give an array of strings with the names of the environment variables on Connect (see [the API Docs](https://docs.posit.co/connect/api/#get-/v1/content/-guid-/environment))

I kept all of the errors bubbling up since the frontend can handle this by ignoring them for the most part and leaving better Connect Content error handling to other, more pointed features.

The plan is to capture these errors and assume that the environment is `[]` if something goes awry. 

## Intent

Part of #2365 

## Type of Change

- - [ ] Bug Fix 
- - [x] New Feature 
- - [ ] Breaking Change 
- - [ ] Documentation 
- - [ ] Refactor 
- - [ ] Tooling 

## Approach

The approach was to fetch everything that is necessary to call the Connect API and fail on the different cases that can happen before calling the API.

Then if we call the API pass through any errors.

## Automated Tests

There are automated tests for the new endpoint.

## Directions for Reviewers

The API can be tested directly using `just run`. There are tests for the various error states:
- If the deployment record with the given name doesn't exist -> `404`
- If the deployment record exists, but is in error and cannot be decoded -> `400`
- If the deployment record exists, but hasn't been deployed -> `400`
- If the deployment record exists, but there is no credential for the given Server URL -> `400`
- Pass through errors from the Connect API
  - For example if authentication fails, or the content has been deleted
- A few `InternalErrors` for unexpected cases

### CODE REVIEW

File: internal/services/api/api_service.go [hunk 1]
```diff
@@ -166,6 +166,10 @@ func RouterHandlerFunc(base util.AbsolutePath, lister accounts.AccountList, log
 	r.Handle(ToPath("deployments", "{name}"), DeleteDeploymentHandlerFunc(base, log)).
 		Methods(http.MethodDelete)
 
+	// GET /api/deployments/$NAME/environment
+	r.Handle(ToPath("deployments", "{name}", "environment"), GetDeploymentEnvironmentHandlerFunc(base, log, lister)).
```
Comment (dotNomad) [inline internal/services/api/api_service.go:170]: Unsure about the name here. Maybe we should be more direct about this being the Connect Content environment?

This is on the `deployment` resource which implies this is the content, and not the configuration's environment.

File: internal/services/api/get_deployment_env.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		if err != nil {
+			// If the deployment file doesn't exist, return a 404
+			if errors.Is(err, fs.ErrNotExist) {
+				http.NotFound(w, req)
+				return
+			}
+			// If the deployment file is in error return a 400
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(fmt.Sprintf("deployment %s is invalid: %s", name, err)))
+			return
```
Comment (marcosnav) [inline internal/services/api/get_deployment_env.go:41]: Many of these error responses can be sent as specific agent errors, like in `post_deployment.go`. I'm guessing those changes are scoped to a future PR, when bridging the frontend, but wanted to confirm.

File: internal/services/api/api_service.go (hunk 1 above)
Comment (marcosnav) [reply internal/services/api/api_service.go:170]: I think it makes sense as you have it 👍

**Review by marcosnav — APPROVED** [review]
LGTM

**Pull request merged**